dbms/src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -271,7 +271,8 @@ BlockInputStreams MergeTreeDataSelectExecutor::readFromParts(
                     part->minmax_idx.parallelogram, data.minmax_idx_column_types))
                 continue;
 
-            if (max_block_number_to_read && part->info.max_block > max_block_number_to_read)
+            auto blocks_iterator = max_blocks_number_to_read.find(part->info.partition_id);
+            if (blocks_iterator != max_blocks_number_to_read.end() && part->info.max_block > blocks_iterator->second)
```

Comment:
Turns out this is incorrect in case of the first insert into a partition. If it is a quorum insert and it is not done yet, the partition will not be present in `max_blocks_number_to_read`. In this case we don't want to include any part from that partition.

Reply:
done